### PR TITLE
Iterator conversions

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: inc/softwareversion.class.php [hunk 1]
```diff
@@ -233,35 +233,45 @@ static function dropdownForOneSoftware($options = []) {
...
       }
 
-      $where = '';
+      // Make a select box
+      $criteria = [
+         'SELECT'    => [
+            'glpi_softwareversions.*',
+            'glpi_states AS name'
+         ],
+         'FROM'      => 'glpi_softwareversions',
```
Comment (orthagh) [inline inc/softwareversion.class.php]: missing DISTINCT

File: inc/stat.class.php [hunk 2]
```diff
@@ -102,71 +101,92 @@ static function getItems($itemtype, $date1, $date2, $type, $parent = 0) {
...
+            $iterator = $DB->request([
+               'SELECT' => ['id', 'name'],
+               'FROM'   => 'glpi_groups',
+               'WHERE'  => [
+                  'OR'  => [
+                     'id'        => $parent,
+                     'groups_id' => $parent
+                  ],
+                  $is_field   => 1
+               ] + getEntitiesRestrictRequest("glpi_groups", '', '', true),
```
Comment (orthagh) [inline inc/stat.class.php]:
```suggestion
               ] + getEntitiesRestrictCriteria("glpi_groups", '', '', true),
```

File: inc/supplier.class.php [hunk 3]
```diff
@@ -477,45 +477,76 @@ function showInfocoms() {
             $linkfield = 'id';
             $itemtable = getTableForItemType($itemtype);
 
-            $query = "SELECT `glpi_infocoms`.`entities_id`, `NAME_FIELD`, `$itemtable`.*
-                      FROM `glpi_infocoms`
-                      INNER JOIN `$itemtable` ON (`$itemtable`.`id` = `glpi_infocoms`.`items_id`) ";
+            $criteria = [
+               'SELECT'       => [],
+               'FROM'         => 'glpi_infocoms',
+               'INNER JOINR'  => [
```
Comment (orthagh) [inline inc/supplier.class.php]:
```suggestion
               'INNER JOIN'  => [
```

File: inc/softwareversion.class.php [hunk 4]
```diff
@@ -233,35 +233,45 @@ static function dropdownForOneSoftware($options = []) {
...
-                              `glpi_states`.`name` AS sname
-                FROM `glpi_softwareversions`
-                LEFT JOIN `glpi_states` ON (`glpi_softwareversions`.`states_id` = `glpi_states`.`id`)
-                WHERE `glpi_softwareversions`.`softwares_id` = '".$p['softwares_id']."'
-                      $where
-                ORDER BY `name`";
-      $result = $DB->query($query);
-      $number = $DB->numrows($result);
+
+      $iterator->request($criteria);
```
Comment (cedric-anne) [inline inc/softwareversion.class.php]:
```suggestion
      $iterator = $DB->request($criteria);
```

**Pull request merged**